collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -867,9 +871,9 @@ public void run() {
                 // We may have jumped to a new index in hierarchy activity, so
                 // refresh
                 break;
-
         }
-        refreshCurrentView();
+
+        // now onResume() gets called and calls refreshCurrentView() to reload current widget
```

Comment:
For the sake of completeness, I think you can add something like "If media is being loaded on a background thread, the refresh will be postponed until the media fully loads."

Reply:
@lognaturel Done